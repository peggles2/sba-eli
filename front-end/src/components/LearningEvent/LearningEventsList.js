import React, { Component } from "react";
import { List } from "semantic-ui-react";
import { connect } from "react-redux";
import LearningEventsItem from "./LearningEventsItem";
import { getLearningEvents } from "../../actions/learningEventActions";

export class LearningEventsList extends Component {
  componentDidMount() {
    this.eventsList();
  }

  eventsList() {
    this.props.dispatch(getLearningEvents(this.props.course_id, this.props.module_id));
  }

  renderEventsList(events = []) {
    if (events.length) {
      return events.map(event => {
        return <LearningEventsItem key={event.id} item={event} />;
      });
    }
    return <li>No data</li>;
  }

  render() {
    return <List>{this.renderEventsList(this.props.learningEvents)}</List>;
  }
}

const mapStateToProps = store => {
  return {
    learningEvents: store.learningEvent.learningEvents
  };
};

export default connect(mapStateToProps)(LearningEventsList)