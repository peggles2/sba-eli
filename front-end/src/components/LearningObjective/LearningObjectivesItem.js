import React from "react";
import { List } from "semantic-ui-react";

const LearningObjectivesItem = (props) => {
  return(
    <List.Item><a href={`/learning_objectives/${props.item.id}`}>{props.item.name}</a></List.Item>
  )
}

export default LearningObjectivesItem;
