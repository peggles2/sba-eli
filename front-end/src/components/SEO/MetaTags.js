import React, { Component } from "react";
import { Helmet } from "react-helmet";

export default class MetaTags extends Component {
  getMetaTitle(pageTitle) {
    return pageTitle
      ? pageTitle
      : "Small Business Administration: Women Entrepreneurs";
  }

  getMetaDescription(description) {
    return description
      ? description
      : "We support America's small businesses. The SBA connects entrepreneurs with lenders and funding to help them plan, start and grow their business.";
  }

  getCanonicalUrl(url) {
    return url && url.startsWith("https://") ? url : "https://sba.gov/eli";
  }

  getMetaImage(imageUrl) {
    return imageUrl ? imageUrl : "https://picsum.photos/200/300/?random";
  }

  render() {
    const metaProps = [
      {
        name: "description",
        content: this.getMetaDescription(this.props.metaDescription)
      },
      {
        property: "og:title",
        content: this.getMetaTitle(this.props.metaTitle)
      },
      {
        property: "og:description",
        content: this.getMetaDescription(this.props.metaDescription)
      },
      {
        property: "og:type",
        content: "website"
      },
      {
        property: "og:image",
        content: this.getMetaImage(this.props.metaImage)
      },
      {
        property: "og:url",
        content: this.getCanonicalUrl(this.props.canonicalUrl)
      }
    ];

    const linkProp = {
      rel: "canonical",
      href: this.getCanonicalUrl(this.props.canonicalUrl)
    };
    return (
      <Helmet
        title={this.getMetaTitle(this.props.metaTitle)}
        meta={metaProps}
        link={linkProp}
      />
    );
  }
}
