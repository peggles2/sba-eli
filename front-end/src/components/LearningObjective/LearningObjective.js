import React, { Component } from "react";
import axios from "axios";
import { Header } from "semantic-ui-react";
import LearningEventsList from "../LearningEvent/LearningEventsList";

export default class LearningObjective extends Component {
  constructor(props) {
    super(props);

    this.state = {
      learningObjective: {}
    };
  }

  componentDidMount() {
    this.learningObjectiveObject();
  }

  learningObjectiveObject() {
    const objective_id = this.props.match.params.id;
    const url =
      process.env.REACT_APP_SERVICE_HOST +
      `/learning_objectives/${objective_id}`;

    const objectiveParams = {
      course_id: this.props.match.params.course_id
    };

    axios
      .get(url, { params: objectiveParams })
      .then(res => {
        console.log("res", res);
        const learningObjective = res.data;

        this.setState({ learningObjective });
      })
      .catch(error => {
        console.log(error);
      });
  }

  render() {
    const learningObjective = this.state.learningObjective;
    return (
      <div>
        <Header as="h1">Learning Objective {learningObjective.id}</Header>
        <Header as="h3">{learningObjective.name}</Header>
        <Header as="h4">Micro Learning Events:</Header>
        <LearningEventsList
          course_id={this.props.match.params.id}
          module_id={this.props.match.params.course_id}
        />
      </div>
    );
  }
}
