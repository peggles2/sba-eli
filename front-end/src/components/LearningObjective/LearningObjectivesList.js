import React from 'react';
import LearningObjectivesItem from './LearningObjectivesItem';

export default class LearningObjectivesList extends React.Component {
    constructor(props) {        
        super(props);
        
        this.state = {
            objectivesList: []
        }                      
    }

    async componentDidMount() {                                       
        const objectivesList = await this.objectivesList()
        this.setState({ objectivesList });
    }
    
    async objectivesList() {        
        const res =  await fetch(
            process.env.REACT_APP_SERVICE_HOST +
            "/learning_objectives/" +
            "?course_id=" +
            this.props.course_id
        )
        
        return await res.json();
    }

    renderObjectivesList(objectives = []) {
        if(objectives.length) {
            return objectives.map(objective => {                
                return(
                    <LearningObjectivesItem key={objective.id} item={objective} />
                )
            })        
        }
        return <li>No data</li>
    }

    render() {               
        return(                        
            <ul>                    
                {this.renderObjectivesList(this.state.objectivesList)}
            </ul>                            
        )
    }
}