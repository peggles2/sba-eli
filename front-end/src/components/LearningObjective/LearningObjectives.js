import React, { Component } from "react";
import LearningObjectivesList from "./LearningObjectivesList";

export default class LearningObjectives extends Component {
  render() {
    return (
      <div>
        <LearningObjectivesList course_id={this.props.match.params.course_id} />
      </div>
    );
  }
}
