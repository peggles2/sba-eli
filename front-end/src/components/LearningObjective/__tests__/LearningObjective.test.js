import React from "react";
import LearningObjective from "../LearningObjective";
import LearningEventsList from "../../LearningEvent/LearningEventsList";
import MetaTags from "../../SEO/MetaTags";
import { shallow, mount } from "enzyme";

describe("LearningObjective", () => {
  it("should render meta tags with Learning Objective specific information", () => {
    const match = { params: { id: 1 } };
    const title = "Learning Objective!"
    const desc = "learning objective description text"
    const learningObjective = { name: title, description: desc};
    
    const wrapper = shallow(<LearningObjective match={match} />);
    wrapper.instance().setState({learningObjective});

    expect(wrapper.find(MetaTags).exists()).toBe(true);
    const metaWrapper = wrapper.find(MetaTags).dive();

    expect(metaWrapper.find('title').text()).toBe(title);
    expect(metaWrapper.find("meta[name='author']").exists()).toBe(false);
    expect(metaWrapper.find("meta[name='description']").exists()).toBe(true);
    expect(metaWrapper.find("meta[name='description']").props().content).toBe(desc);
    //TODO: add Canonical URL validation when we have final urls
  });

  it("should render a single h1 and it should display the objective title", () => {
    const match = { params: { id: 1 } };
    const title = "Learning Objective!"
    const desc = "learning objective description text"
    const learningObjective = { name: title, description: desc};
    
    const wrapper = mount(<LearningObjective match={match} />);
    wrapper.instance().setState({learningObjective});

    expect(wrapper.find("h1").exists()).toBe(true);
    expect(wrapper.find("h1").length).toEqual(1);
    expect(wrapper.find("h1").text()).toEqual(title);
  });

  it("should render a <LearningEventsList>", () => {
    const match = { params: { id: 1 } };
    const wrapper = mount(<LearningObjective match={match} />);

    expect(wrapper.find(LearningEventsList).exists()).toBe(true);
  });
});
