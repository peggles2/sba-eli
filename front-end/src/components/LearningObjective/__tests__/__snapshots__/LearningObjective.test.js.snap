// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningObjective should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningObjective
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
        >
          Learning Objective 
        </Header>,
        <Header
          as="h3"
        />,
        <Header
          as="h4"
        >
          Micro Learning Events:
        </Header>,
        <LearningEventsList
          course_id={1}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": Array [
            "Learning Objective ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Learning Objective ",
          undefined,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h3",
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h4",
          "children": "Micro Learning Events:",
        },
        "ref": null,
        "rendered": "Micro Learning Events:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "course_id": 1,
          "module_id": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
          >
            Learning Objective 
          </Header>,
          <Header
            as="h3"
          />,
          <Header
            as="h4"
          >
            Micro Learning Events:
          </Header>,
          <LearningEventsList
            course_id={1}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": Array [
              "Learning Objective ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Learning Objective ",
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h3",
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h4",
            "children": "Micro Learning Events:",
          },
          "ref": null,
          "rendered": "Micro Learning Events:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "course_id": 1,
            "module_id": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
